### PR TITLE
Set tootips as default label component for common charts

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

Set tootips as default label component for charts that do not have labels/legends. These include:

 - [x] Bar chart,
 - [x] Line chart, and
 - [x] Pie chart.

Also
 - [x] Ensure this prop is exposed and override-able/customizable.

NOTE:

The following charts use custom svg to draw parts of their charts and will require custom implementation (at least custom event handling to show tooltip). They will be implemented in the next iteration (where we'll also implement multiple fill/font styles for a tooltip):

 - Bullet chart, and
 - Nested proportional are charts,

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots

Peek 2019-08-09 13-51

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by karimkawambwa** [comment]
@kilemensi resolve the conflicts then I can take a look at this.

**Comment by kilemensi** [comment]
> @kilemensi resolve the conflicts then I can take a look at this.

Will do @karimkawambwa but with the new designs, I'm going to have to take this back to Draft/WIP.

**Comment by kilemensi** [comment]
Think Tooltip MVP is ready @karimkawambwa / @nerdlet .

File: stories/chart.stories.tsx [hunk 1]
```diff
@@ -222,7 +245,13 @@ storiesOf('HURUmap UI|Charts/LineChart', module)
               }
             }
           },
-          scatter: [{ size: 5, symbol: 'circle' }, { size: 5, symbol: 'plus' }]
+          group: {
+            // Line chart combines line and scatter hence best to define
+            // labels at group level
+            labels: datum => `${datum.x}\n${datum.geo} ${datum.y}`
```
Comment (karimkawambwa) [inline stories/chart.stories.tsx]: Does this mean we can't have normal labels on the points or above a bar in a bar chart?

File: stories/chart.stories.tsx [hunk 2]
```diff
@@ -80,15 +92,26 @@ storiesOf('HURUmap UI|Charts/BarChart', module)
         <BarChart
           width={number('width', 500)}
           height={number('height', 300)}
-          labels={datum => `${datum.y}${text('dataUnit', '%')}`}
+          labels={datum =>
+            `Group ${datum.tick} Tick\n${datum.x} ${datum.y}${text(
+              'dataUnit',
+              '%'
+            )}`
+          }
```
Comment (karimkawambwa) [inline stories/chart.stories.tsx]: So no more labels support ontop of bars ?

File: src/NestedProportionalAreaChart/index.tsx [hunk 3]
```diff
@@ -95,6 +95,7 @@ function NestedProportionalAreaChart({
               data={data}
               groupSpacing={groupSpacing}
               labels={() => ''} // Don't show PieChart labels
+              labelComponent={undefined} // Don't show PieChart labels
```
Comment (karimkawambwa) [inline src/NestedProportionalAreaChart/index.tsx]: Which of the two really works ?

Comment (kilemensi) [reply src/NestedProportionalAreaChart/index.tsx]: The second completely removes the component (necessary for VictoryTooltip)... The above was good enough when we were using VictoryLabel

File: stories/chart.stories.tsx (hunk 1 above)
Comment (kilemensi) [reply stories/chart.stories.tsx]: I don't see labels at the top in the designs (both Dominion and new HURUmap Designs),

----

I haven't tested it but may be defining `labelComponent` on the bar chart itself may allow extra labels.

File: stories/chart.stories.tsx (hunk 2 above)
Comment (kilemensi) [reply stories/chart.stories.tsx]: Answered above.

**Pull request merged**